forester/src/epoch_manager.rs

```diff
@@ -203,22 +152,16 @@ impl<R: RpcConnection, I: Indexer<R>> EpochManager<R, I> {
             let next_epoch = current_epoch + 1;
             let next_phases = get_epoch_phases(&self.protocol_config, next_epoch);
             let mut rpc = self.rpc_pool.get_connection().await?;
-            let mut slot = rpc.get_slot().await?;
             let slots_to_wait = next_phases.registration.start.saturating_sub(slot);
             info!(
                 "Waiting for epoch {} registration phase to start. Current slot: {}, Registration phase start slot: {}, Slots to wait: {}",
                 next_epoch, slot, next_phases.registration.start, slots_to_wait
             );
-            let sleep_duration = Duration::from_millis(400 * slots_to_wait);
-            debug!("Sleeping for {} ms", sleep_duration.as_millis());
-            tokio::time::sleep(sleep_duration).await;
-            while slot < next_phases.registration.start {
-                tokio::time::sleep(Duration::from_millis(400)).await;
-                slot = rpc.get_slot().await?;
-                debug!(
-                    "Current slot: {}, Registration phase start slot: {}",
-                    slot, next_phases.registration.start
-                );
+            if let Err(e) =
+                wait_until_slot_reached(&mut *rpc, next_phases.registration.start, 400).await
```

Comment:
also commented this in a different pr, let's make 400 a constant or part of the `SlotTracker`.
From my end, doesn't need to be fixed in this pr if it doesn't make sense.